Documentation/spec/docfx_document_schema.md

```diff
@@ -0,0 +1,197 @@
+# DocFX Document Schema Design Specification
+
+## 1. Introduction
+DocFX supports different [document processors](..\tutorial\howto_build_your_own_type_of_documentation_with_custom_plug-in.md) to handle different kinds of input. For now, if the data model changes a bit, a new document processor is needed, even most works in processors are the same.
+
+DocFX Document Schema (abbreviated to `this schema` below) is introduced to address ths problem. This schema is a JSON media type for defining the structure of a DocFX document. This schema is intended to define manipulation, documentation, validation of the document data.
+
+## 2. Conventions and Terminology
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## 3. Overview
+DocFX Document Schema is in [JSON](http://www.json.org/) format. It borrows most syntax from [JSON Schema](http://json-schema.org/), while it also introduces some other syntax to manuplating the data.
+
+### 3.1 Validation
+This schema can describe the structure of a DocFX document. 
+[JSON schema validation](http://json-schema.org/latest/json-schema-validation.html) already defines many keywords. This schema starts from supporting limited keyword like `type`, `properties`.
+
+### 3.2 Manipulation
+This schema can describe how to handle each property of the document.
+
+Some handler is predefined in DocFX, like markup a string using [DFM](..\spec\docfx_flavored_markdown.md), mark a property as an uid, include content from another source, etc. 
+
+Some handler can be injected into the schema parsing process to accomplish some custom steps.
+
+### 3.3 Integration with DocFX
+Schema is passed to DocFX as part of the template. It MUST be put in the `schemas` sub folder under template folder. The file name is treated as the schema name, also known as the document type of a template.
+
+For example, to add a schema for a document type `ManagedReference`, a schema file should added like `{templateFolder}\schemas\ManagedReference.json`.
+
+Schema describes the object model. The format of the object model is not constrained. YAML and JSON should both be supported.
```

Comment:
will remove this line and only mention at start...

---
In reply to: [122871077](https://github.com/dotnet/docfx/pull/1771#discussion_r122871077) [](ancestors = 122871077)